### PR TITLE
Allow self_validate config to be a String mapping to the desired stage

### PR DESCRIPTION
**Pull request by kevmoo**
Allows other jobs to run in parallel while validate executes

### CODE REVIEW

File: mono_repo/lib/src/mono_config.dart [hunk 1]
```diff
@@ -218,6 +218,24 @@ class MonoConfig {
...
+    return value ? _selfValidateStageName : null;
+  }
+
+  if (value is String) {
+    return value;
+  }
+
+  throw ArgumentError.value(value, 'value', 'Must be a `String` or `bool`.');
+}
+
```
Comment (jakemac53) [inline mono_repo/lib/src/mono_config.dart:238]:
```suggestion
String _selfValidateFromValue(Object value) {
  if (value == null) return null;
  if (value is bool) return value ? _selfValidateStageName : null;
  if (value is String) return value;
  throw ArgumentError.value(value, 'value', 'Must be a `String` or `bool`.');
}

```

File: mono_repo/mono_pkg.yaml [hunk 2]
```diff
@@ -1,11 +1,10 @@
 # See https://github.com/google/mono_repo.dart for details
 
 stages:
-- mono_repo_self_validate:
-    # A bit weird, this should really be a global task but we don't have those.
-    - command: cd ../ && dart mono_repo/bin/mono_repo.dart travis --validate
-      dart: dev
 - smoke_test:
+  # https://github.com/google/mono_repo.dart/issues/199
```
Comment (jakemac53) [inline mono_repo/mono_pkg.yaml:5]: shouldn't this use the new self validate stuff?

Comment (kevmoo) [reply mono_repo/mono_pkg.yaml:5]: Then I'd have to wire up all of this crazy to detect itself and then activate itself from `path` etc – too much crazy!

**Comment by kevmoo** [comment]
@jakemac53 – LGTY?

**Review by jakemac53 — APPROVED** [review]

**Pull request merged**